Match/IMatchCycle.cs

```diff
@@ -0,0 +1,18 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Match
+{
+    public interface IMatchCycle
+    {
+        /// information on what features the match implementation supports
+        public FeatureSet FeatureSet { get; }
+
+        /// Sets up a match for the provided match info,
+        /// and returns a delegate to perform the prepared match.
+        public Task<Perform> SetUp(MatchInfo matchInfo, CancellationToken? token = null);
+
+        /// A delegate that performs a match that was prepared with <see cref="IMatchCycle.SetUp"/>.
+        delegate Task<MatchResult> Perform(CancellationToken? token = null);
```

Comment:
delegate vs event?

Reply:
not sure what you mean

Reply:
The reason this isn't just a second interface method is to make sure nobody could call Perform before SetUp

Reply:
I mean is an event preferable to a delegate here

Reply:
after a lengthly discord discussion we agreed that this is confusing and should just be two interface methods